docs/module-sources.rst

```diff
@@ -0,0 +1,562 @@
+        - type: file
+          path: org.example.foo.metainfo.xml
+
+    - name: unsquashfs
+      buildsystem: simple
+      build-commands:
+       - XZ_SUPPORT=1 make -C squashfs-tools -j ${FLATPAK_BUILDER_N_JOBS} unsquashfs
+       - install -Dpm755 -t "${FLATPAK_DEST}/bin" squashfs-tools/unsquashfs
+      sources:
+        - type: git
+          url: https://github.com/plougher/squashfs-tools.git
+          tag: 4.6.1
+          commit: d8cb82d9840330f9344ec37b992595b5d7b44184
+
+    - name: bar
+      buildsystem: simple
+      build-commands:
+       - unsquashfs -dest bar -quiet -no-progress squashed.snap
+       - mv bar ${FLATPAK_DEST}
+       - ln -s ${FLATPAK_DEST}/bar/executable ${FLATPAK_DEST}/bin/executable
+      sources:
+        - type: file
+          dest-filename: squashed.snap
+          url: https://example.org/squashed-1.0.snap
+          sha256: <sha256 checksum of squashed-1.0.snap>
+
+
+``dest-filename`` is typically used with this source type to make the
+filename `predictable` for the commands in the ``build-commands`` array.
+
+Patch sources
+-------------
+
+These are sources of ``type: patch`` used with patchfiles. The typical
+usage is to create patchfiles and place them somewhere relative to the
+manifest root. The ``path`` property is the location of the file relative
+to the manifest.
+
+.. code-block:: yaml
+
+  modules:
+    - name: foo
+      buildsystem: cmake-ninja
+      builddir: true
+      config-opts:
+      - ...
+    sources:
+      - type: git
+        url: https://example.org/foo.git
+        tag: v1.0.0
+        commit: 63c01f14391aef7d7691c7c63a610d47512147ed
+      - type: patch
+        path: patches/test-patch.patch
+
+Multiple patches can be specified with ``paths``:
+
+.. code-block:: yaml
+
+  - type: patch
+    paths:
+      - patches/test1-patch.patch
+      - patches/test2-patch.patch
+
+By default, ``patch -p 1`` is used, but it can be adjusted with
+``strip-components`` and additional options can be passed with
+``options`` :
+
+.. code-block:: yaml
+
+  - type: patch
+    paths:
+      - patches/test1-patch.patch
+      - patches/test2-patch.patch
+    options:
+      - -d
+      - dir1
+
+``use-git: true`` and ``use-git-am: true`` can be used to use
+``git apply``and ``git am`` respectively instead of ``patch``.
+
+Shell and Script sources
+------------------------
+
+A shell source with ``type: shell`` is used to modify a source during
+extraction. ``commands`` takes in any shell commands.
+
+This is usually used to make quick modifications to sources but a patch
+should be used for any complex modification to make it fail safe.
+
+.. code-block:: yaml
+
+  modules:
+    - name: foo
+      sources:
+        - type: archive
+          url: https://example.org/foo.tar.xz
+          sha256: ea029a2e21d2d6ad0a156f6679bd66836204aa78148a4c5e498fe682e77127ef
+        - type: shell
+          commands:
+            - sed -i 's|/usr/local|/app|g' makefile.common
```

Comment:
I'd rather use the example of the autoconf files. Using `sed` should be discouraged in general.